### PR TITLE
Migrate users components to typescript

### PR DESCRIPTION
**Pull request by kmerz**
Convert all User related Components, Stores and Logic to Typescript.

### CODE REVIEW

File: graylog2-web-interface/src/components/common/PaginatedItemOverview/PaginatedItemOverview.tsx [hunk 1]
```diff
@@ -42,15 +41,15 @@ type Props = {
...
+  onDeleteItem?: (descriptiveItem: DescriptiveItem) => void,
+  queryHelper?: React.ReactNode,
 };
 
 const pageSizes = [5, 10, 30];
 export const DEFAULT_PAGINATION = { page: INITIAL_PAGE, perPage: pageSizes[0], query: '' };
 
 const PaginatedItemOverview = ({ onLoad, overrideList, onDeleteItem, queryHelper, noDataText }: Props) => {
-  const [paginatedList, setPaginatedList] = useState<?PaginatedListType>();
+  const [paginatedList, setPaginatedList] = useState<PaginatedListType>();
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/common/PaginatedItemOverview/PaginatedItemOverview.tsx:52]: IMO this should be:
```suggestion
  const [paginatedList, setPaginatedList] = useState<PaginatedListType | undefined>();
```

File: graylog2-web-interface/src/components/common/PaginatedItemOverview/PaginatedItemOverview.tsx [hunk 2]
```diff
@@ -89,7 +89,7 @@ const PaginatedItemOverview = ({ onLoad, overrideList, onDeleteItem, queryHelper
                   queryHelpComponent={queryHelper}
                   searchButtonLabel="Filter" />
       <div>
-        {itemList}
+        {itemList !== undefined ? itemList : emptyResult}
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/common/PaginatedItemOverview/PaginatedItemOverview.tsx:92]: We can also do a
```suggestion
        {itemList ?? emptyResult}
```

File: graylog2-web-interface/src/components/users/TimeoutInput.tsx [hunk 3]
```diff
@@ -81,6 +80,8 @@ const TimeoutInput = ({ value: propsValue, onChange }: Props) => {
     setUnit(newUnit);
   };
 
+  // @ts-ignore
+  // @ts-ignore
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/users/TimeoutInput.tsx:84]: I think these slipped in through IntelliJ?

File: graylog2-web-interface/src/components/users/UserCreate/TimeoutFormGroup.tsx [hunk 4]
```diff
@@ -24,9 +23,6 @@ const TimeoutFormGroup = () => (
   <Field name="session_timeout_ms">
     {({ field: { name, value, onChange } }) => (
       <TimeoutInput value={value}
-                    labelSize={3}
-                    controlSize={9}
-                    name={name}
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/users/UserCreate/TimeoutFormGroup.tsx:29]: Why are these removed?

File: graylog2-web-interface/src/components/users/UserCreate/UserCreate.tsx [hunk 5]
```diff
@@ -60,11 +59,13 @@ const _validate = (values) => {
   return errors;
 };
 
+type RequestError = { additional: { res: { text: string }}};
+
 const UserCreate = () => {
   const initialRole = { name: 'Reader', description: 'Grants basic permissions for every Graylog user (built-in)', id: '' };
-  const [users, setUsers] = useState();
+  const [users, setUsers] = useState<Immutable.List<User>>();
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/users/UserCreate/UserCreate.tsx:66]: Shouldn't this be
```suggestion
  const [users, setUsers] = useState<Immutable.List<User> | undefined>();
```

File: graylog2-web-interface/src/components/users/UsersOverview/UsersOverview.tsx [hunk 6]
```diff
@@ -86,11 +87,11 @@ const _updateListOnUserDelete = (perPage, query, setPagination) => UsersActions.
 const _updateListOnUserSetStatus = (pagination, setLoading, setPaginatedUsers) => UsersActions.setStatus.completed.listen(() => _loadUsers(pagination, setLoading, setPaginatedUsers));
 
 const UsersOverview = () => {
-  const currentUser = useContext(CurrentUserContext);
-  const [paginatedUsers, setPaginatedUsers] = useState<?PaginatedUsers>();
+  const currentUser = useContext<UserJSON>(CurrentUserContext);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/users/UsersOverview/UsersOverview.tsx:90]: Should the state value be optional?

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good, added some small comments inline.

File: graylog2-web-interface/src/components/users/TimeoutInput.tsx (hunk 3 above)
Comment (kmerz) [reply graylog2-web-interface/src/components/users/TimeoutInput.tsx:84]: yes :smile:

File: graylog2-web-interface/src/components/users/UserCreate/TimeoutFormGroup.tsx (hunk 4 above)
Comment (kmerz) [reply graylog2-web-interface/src/components/users/UserCreate/TimeoutFormGroup.tsx:29]: Because they are not used anymore in TimeoutInput.

**Review by dennisoelkers — APPROVED** [review]
LGTM!
